adxl345/adxl345.go

```diff
@@ -0,0 +1,230 @@
+	S16G Sensitivity = 0x03 // Sensitivity at 16g
+
+	ThreshTap        = 0x1D // Tap threshold
+	OfsX             = 0x1E // X-axis offset
+	OfsY             = 0x1F // Y-axis offset
+	OfsZ             = 0x20 // Z-axis offset
+	Dur              = 0x21 // Tap duration
+	Latent           = 0x22 // Tap latency
+	Window           = 0x23 // Tap window
+	ThreshAct        = 0x24 // Activity threshold
+	ThreshInact      = 0x25 // Inactivity threshold
+	TimeInact        = 0x26 // Inactivity time
+	ActInactCtl      = 0x27 // Axis control for activity/inactivity detection
+	ThreshFf         = 0x28 // Free-fall threshold
+	TapAxes          = 0x2A // Axis control for single tap/double tap
+	TapStatus        = 0x2B // Source of single tap/double tap
+	ActivityStatus   = 0x2A // Source of activity detection
+	InactivityStatus = 0x2B // Source of inactivity detection
+
+	// Control registers
+
+	BwRate     = 0x2C // Data rate and power mode control
+	PowerCtl   = 0x2D // Power saving features control
+	IntEnable  = 0x2E // Interrupt enable control
+	IntMap     = 0x2F // Interrupt mapping control
+	IntSource  = 0x30 // Source of interrupts
+	DataFormat = 0x31 // Data format control
+
+	// Data registers
+	DataX0 = 0x32 // X-Axis Data 0
+	DataX1 = 0x33 // X-Axis Data 1
+	DataY0 = 0x34 // Y-Axis Data 0
+	DataY1 = 0x35 // Y-Axis Data 1
+	DataZ0 = 0x36 // Z-Axis Data 0
+	DataZ1 = 0x37 // Z-Axis Data 1
+
+	// FIFO control
+	FifoCtl    = 0x38 // FIFO control
+	FifoStatus = 0x39 // FIFO status
+
+)
+
+// DefaultSpiFrequency is the default SPI frequency used to communicate with the device.
+var (
+	SpiFrequency = physic.KiloHertz * 50
+	SpiMode      = spi.Mode3 // Defines the base clock signal, along with the polarity and phase of the data signal.
+	SpiBits      = 8
+)
+
+var DefaultOpts = Opts{
+	TurnOnOnStart:    true,
+	ExpectedDeviceID: 0xE5,
+	Sensitivity:      S2G,
+}
+
+type Opts struct {
+	TurnOnOnStart    bool        // Turn on the device in measurement mode on start.
+	ExpectedDeviceID byte        // Expected device ID used to verify that the device is an ADXL345.
+	Sensitivity      Sensitivity // Sensitivity of the device (2G, 4G, 8G, 16G)
+}
+
+// Dev is a driver for the ADXL345 accelerometer
+// It uses the SPI interface to communicate with the device.
+type Dev struct {
+	name        string
+	s           spi.Conn
+	sensitivity Sensitivity
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("ADXL345{Sensitivity:%s}", d.sensitivity)
+}
+
+// New creates a new ADXL345 Dev or returns an error.
+// The bus and chip parameters define the SPI bus and chip select to use.
+// The SPI s is configured.
+// The device is turned on.
+// The device is verified to be an ADXL345.
+func New(p spi.Port, o *Opts) (*Dev, error) {
+	// Convert the spi.Port into a spi.Conn so it can be used for communication.
+	c, err := p.Connect(SpiFrequency, SpiMode, SpiBits)
+	if err != nil {
+		return nil, err
+	}
+	d := &Dev{
+		name: "ADXL345",
+		s:    c,
+	}
+	if o.TurnOnOnStart {
```

Comment:
I don't think it's necessary, it's okay to have users call it manually, or always do it by default.

Reply:
Removed.